scripts/local_env_create_vm

```diff
@@ -0,0 +1,137 @@
+
+$(tput sgr0)"
+
+    COLS="$(echo "$CONTENT" | wc -L)"
+    printf '%*s' "${COLS}" '' | tr ' ' \#
+    printf "$CONTENT"
+    printf '%*s\n' "${COLS}" '' | tr ' ' \#
+}
+
+if [ "$EUID" -ne 0 ]; then
+    echo "Local env needs sudo permissions"
+    exit 1
+fi
+
+SCRIPTPATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+
+CIFMW_LOCAL_ENV_BASE_DIR="${1}"
+CIFMW_LOCAL_ENV_BACKING_IMG_URL="${2}"
+CIFMW_LOCAL_ENV_PULL_SECRET_PATH="${3}"
+CIFMW_LOCAL_ENV_CALLING_USER=$([ -z "$SUDO_USER" ] && echo "root" || echo "$SUDO_USER")
+CIFMW_LOCAL_ENV_CALLING_USER_HOME=$(eval echo "~$CIFMW_LOCAL_ENV_CALLING_USER")
+
+CIFMW_LOCAL_ENV_IMAGES_DIR="${1}/images"
+CIFMW_LOCAL_ENV_BACKING_IMG="${CIFMW_LOCAL_ENV_IMAGES_DIR}/$(basename ${CIFMW_LOCAL_ENV_BACKING_IMG_URL})"
+CIFMW_LOCAL_ENV_VM_IMG="${CIFMW_LOCAL_ENV_IMAGES_DIR}/ci-cs9.qcow2"
+CIFMW_LOCAL_ENV_VM_NAME="ci-cs9-vm"
+
+if [[ ! -f "$CIFMW_LOCAL_ENV_PULL_SECRET_PATH" ]]; then
+    echo "CIFMW_LOCAL_ENV_PULL_SECRET_PATH must point to an existing pull-secret"
+    exit 1
+fi
+
+if [[ ! -f "$CIFMW_LOCAL_ENV_BACKING_IMG" ]]; then
+    mkdir -p "$CIFMW_LOCAL_ENV_IMAGES_DIR"
+    curl -l -o "$CIFMW_LOCAL_ENV_BACKING_IMG" "${CIFMW_LOCAL_ENV_BACKING_IMG_URL}"
+    curl -l -o "${CIFMW_LOCAL_ENV_BACKING_IMG}.SHA256SUM" "${CIFMW_LOCAL_ENV_BACKING_IMG_URL}.SHA256SUM"
+    (cd "$CIFMW_LOCAL_ENV_IMAGES_DIR" ; sha256sum -c "${CIFMW_LOCAL_ENV_BACKING_IMG}.SHA256SUM")
+fi
+
+rm -rf "$CIFMW_LOCAL_ENV_VM_IMG"
+
+qemu-img create -o backing_file="$CIFMW_LOCAL_ENV_BACKING_IMG",backing_fmt=qcow2 -f qcow2 "$CIFMW_LOCAL_ENV_VM_IMG" "120G"
+virt-sysprep -a "$CIFMW_LOCAL_ENV_VM_IMG" --network \
+  --ssh-inject root:file:${CIFMW_LOCAL_ENV_CALLING_USER_HOME}/.ssh/id_rsa.pub \
+  --selinux-relabel \
+  --root-password "password:fooBar" \
+  --firstboot "${SCRIPTPATH}/local_env_vm_first_boot" \
+  --update \
+  --write "/root/pull-secret:$(cat ${CIFMW_LOCAL_ENV_PULL_SECRET_PATH})" \
+  --install vim,git,tmux,libvirt,libvirt-client,libvirt,libvirt-daemon-kvm,virt-install,qemu-kvm,make,bash-completion,guestfs-tools,libguestfs,python3-pip
+
+virt-install --import \
+--name "$CIFMW_LOCAL_ENV_VM_NAME" \
+--memory 22572 \
+--vcpus 6 \
+--cpu host \
+--os-variant=centos-stream9 \
+--graphics spice \
+--noautoconsole \
+--network=default \
+--disk "$CIFMW_LOCAL_ENV_VM_IMG"
+
+CIFMW_LOCAL_ENV_VM_IP_CNT=0
+echo "Waiting for VM to come up"
+until [[ "$CIFMW_LOCAL_ENV_VM_IP_CNT" -ge 5 ]]
+do
+    CIFMW_LOCAL_ENV_VM_IP=$(virsh -q domifaddr --domain "$CIFMW_LOCAL_ENV_VM_NAME" | awk -F ' ' '{print $4}' | cut -d "/" -f 1)
+    if [[ ! -z $CIFMW_LOCAL_ENV_VM_IP ]]; then
+        break
+    fi
+    CIFMW_LOCAL_ENV_VM_IP_CNT=$((CIFMW_LOCAL_ENV_VM_IP_CNT+1))
+    sleep 5
+done
+
+if [ -z "$CIFMW_LOCAL_ENV_BACKING_IMG" ]; then
+    echo "Unable to find the $CIFMW_LOCAL_ENV_VM_NAME IP"
+    exit 1
+fi
+
+# Copy local ci-framework to target
+rsync -r -l \
+-e "ssh -i ${CIFMW_LOCAL_ENV_CALLING_USER_HOME}/.ssh/id_rsa -o StrictHostKeyChecking=no" \
```

Comment:
maybe adding a block in the ~/.ssh/config would make it more convenient for future use?